analyzer/consensus/consensus.go

```diff
@@ -323,8 +328,33 @@ func (m *Main) processBlock(ctx context.Context, height int64) error {
 	return nil
 }
 
+// getTxHashIndexLookup returns the array of block transactions.
+func (m *Main) getTxHashIndexLookup(ctx context.Context, height int64) (map[string]int, error) {
+	txHashIndexLookup := make(map[string]int)
+
+	source, err := m.source(height)
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := source.BlockData(ctx, height)
+	if err != nil {
+		return nil, err
+	}
+
+	for i := range data.Transactions {
+		_, exists := txHashIndexLookup[data.Transactions[i].Hash().Hex()]
+		// Event should associated with the first transaction of this hash.
```

Comment:
not so much "should," but it's what we're stuck with unless we change oasis-core

Reply:
Agree.